authors/apps/articles/tests/api/test_reactions.py

```diff
@@ -0,0 +1,204 @@
+from django.urls import reverse
+from rest_framework import status
+
+from authors.apps.articles.models import Article
+from authors.apps.articles.tests.api.test_articles import BaseArticlesTestCase
+from authors.apps.core.test_helpers import set_test_client, login, logout
+
+
+class BaseReactionsTestCase(BaseArticlesTestCase):
+    def setUp(self):
+        super().setUp()
+        set_test_client(self.client)
+        self.slug = self.generate_article()
+        logout()
+
+    # like helpers
+    def like_url(self, slug):
+        return reverse('articles:like', kwargs={'slug': slug})
+
+    def like(self, slug):
+        return self.client.post(self.like_url(slug=slug))
+
+    def un_like(self, slug):
+        return self.client.delete(self.like_url(slug=slug))
+
+    # dislike helpers
+    def dislike_url(self, slug):
+        return reverse('articles:dislike', kwargs={'slug': slug})
+
+    def dislike(self, slug):
+        return self.client.post(self.dislike_url(slug=slug))
+
+    def un_dislike(self, slug):
+        return self.client.delete(self.dislike_url(slug=slug))
+
+    # reaction helpers
+    def get_reactions(self, slug):
+        return self.client.get(reverse('articles:reactions', kwargs={'slug': slug}))
+
+    def assertReactionsEqual(self, reactions, like_count=0, like_me=False, dislike_count=0, dislike_me=False):
+        """
+        Asserts that the passed reactions dictionary's fields
+        are equal to those supplied by the other arguments.
+        :param reactions: dict
+        :param like_count: int
+        :param like_me: bool
+        :param dislike_count: int
+        :param dislike_me: bool
+        """
+        self.assertEqual(reactions, {
+            'likes': {
+                'count': like_count,
+                'me': like_me
+            },
+            'dislikes': {
+                'count': dislike_count,
+                'me': dislike_me
+            }
+        })
+
+    def generate_article(self, field='slug', return_obj=False):
+        """
+        Creates an article and returns the field specified by the field
+        argument. If return_obj is set to True is returns the created
+        article object instead of the specified field.
+        :param field:
+        :param return_obj:
+        :return:
+        """
+        article = self.create_article()
+
+        if return_obj:
+            return Article.objects.get(slug=article['slug'])
+
+        return article[field]
+
+
+class ReactionsTestCase(BaseReactionsTestCase):
+    def setUp(self):
+        super().setUp()
+
+    def test_users_can_get_reactions(self):
+        response = self.get_reactions(self.slug)
+        self.assertReactionsEqual(response.data['reactions'])
+
+    def test_like_count_is_accurate(self):
+        login()
+        self.like(self.slug)
```

Comment:
if `generate_article()` returns an article, `self.slug` will be an article right? `self.like` takes in a user, not an article if I am right.

Reply:
I get what is happening but I would suggest you use better naming of the variables.

Reply:
`generate_article()` creates an article and returns its slug.

Reply:
It is returning the slug because the `slug` is the field that is used to reference an `article` on the API. e.g. `api/articles/<str:slug>/like`